shared/src/pool_cache.rs

```diff
@@ -103,6 +126,21 @@ impl PoolCache {
         }
         Ok(())
     }
+
+    // Sometimes nodes requests error when we try to get state from what we think is the current
+    // block when the node has been load balanced out to one that hasn't seen the block yet. As a
+    // workaround we repeat the request up to N times while sleeping in between.
+    async fn fetch_inner(&self, pairs: HashSet<TokenPair>, block: Block) -> Result<Vec<Pool>> {
+        let fetch = || self.inner.fetch(pairs.clone(), block);
+        for _ in 0..self.maximum_retries {
+            match fetch().await {
+                Ok(pools) => return Ok(pools),
+                Err(err) => tracing::debug!("retrying pool fetch because error: {:?}", err),
```

Comment:
Super glad this wont be an logged error anymore! Cheers to quiet, but informative alert channels.

Reply:
It should still be an error if all the retries fail.